static-checks/nist-validation/test.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/python3
+
+import re
+import requests
+from pathlib import Path
+
+from lib import util, results
+
+
+# Get scapval zip file link from NIST web
+scapval_web = 'https://csrc.nist.gov/Projects/Security-Content-Automation-Protocol/SCAP-Releases/SCAP-1-3' # noqa
+r = requests.get(scapval_web)
+scapval_web_path = re.search(r'href="(.*scapval-.*.zip)"', r.text)
+if not scapval_web_path:
+    raise RuntimeError("SCAPval tool link not found on NIST web")
+scapval_link = 'https://csrc.nist.gov' + scapval_web_path.group(1)
+scapval_zip = scapval_link.split('/')[-1]
+
+with util.get_content(build=False) as content_dir:
+    util.subprocess_run(['wget', scapval_link], check=True)
+    util.subprocess_run(['unzip', scapval_zip], check=True)
+
+    ssg_scapval_script = content_dir / 'tests' / 'run_scapval.py'
+    scapval_jar = next(Path.cwd().glob('scapval-*.jar'))
```

Comment:
Does it really make sense to use `run_scapval.py` instead of `scapval.sh` directly in the unzipped tool? ... From what I can tell, `run_scapval.py` seems to be mostly about DS version 1.2 vs 1.3, and parsing some results according to some external (XML?) namespace ... ?

Couldn't we just do something like this?
```python
for ds in util.iter_datastreams():
    ds_name = ds.stem
    result_file = f'{ds_name}.result.xml'
    report_file = f'{ds_name}.report.html'
    proc = util.subprocess_run([
        './scapval.sh', .....,
        '-valresultfile', result_file,
        '-valreportfile', report_file,
        '-file', ds
    ])
    status = 'pass' if proc.returncode == 0 else 'fail'
    results.report(status, name=ds_name, logs=[result_file, report_file])
```
Personally, I think I would prefer reporting each DS from our code, rather than relying on parsing external format. I'm sure this came up in some other test too.

Reply:
Note that I assumed `scapval` would care about exit code, I didn't actually check it. If you end up relying on it, please test it with a malformed XML, to ensure it exits non-0.